src/Microsoft.VisualBasic/tests/VB/TextFieldParserTests.vb

```diff
@@ -0,0 +1,211 @@
+Option Explicit On
+Option Strict On
+
+Imports System.Runtime.CompilerServices
+Imports Microsoft.VisualBasic.FileIO
+Imports Xunit
+
+Namespace Microsoft.VisualBasic.Tests.VB
+    Public NotInheritable Class TextFieldParserTests
+        Inherits IO.FileCleanupTestBase
+
+        Private Const HeaderLine As String = "seq,oneword,age,street,city,state,zip,dollar,pick,date,paragraph"
+
+        Private Const CSVData As String =
+                    HeaderLine & vbCrLf &
+            "1,Test,99,Two Words,Rumnoman,KS,80586,$8333.85,GREEN,09/28/1929,""Ipmadaw udoteac vu ote ozehop ane lujat, lar buh cugi lef owici tat liz cogde.
+Fatigiug kojitfu taapi iz alujok me zimipa koz latkekij vem fo si cepzizhub li por ejcirlu.
+Ewujajlu mivec tooju ad bu cowic irtafit ehaoca vojhehfo aztidun zo wezecmo abe muz wikhutwen idce.
+Vesu sejawga tef lahi dirueg si uwmac bidiw nowidza daime sapmim ki casdun urokir tawdac rahaw beiweed."""
+
+        Private Shared ReadOnly FixedFieldWidths() As Integer = {5, 10, 6, -1}
+        Private Shared ReadOnly FixedFormatData() As String = {
+            "Err  1001                  Cannot access resource.",
+            "Err  2014                  Resource not found.",
+            "Acc  10/03/2009 User1      Administrator.",
+            "Err  0323                  Warning: Invalid access attempt.",
+            "Acc  10/03/2009 User2      Standard user.",
+            "Acc  10/04/2009 User2      Standard user."
+        }
+
+        Private Shared ReadOnly ErrorFormat As Integer() = {5, 5, -1}
+        Private Shared ReadOnly StdFormat As Integer() = {5, 10, 11, -1}
+
+        Enum FormatType
+            ErrorFormat = 0
+            StdFormat = 1
+        End Enum
+
+        Private Shared ReadOnly MultipleFormatData() As String = {
+            "Err  1001 Cannot access resource.",
+            "Err  2014 Resource not found.",
+            "Acc  10/03/2009User1      Administrator.",
+            "Err  0323 Warning: Invalid access attempt.",
+            "Acc  10/03/2009User2      Standard user.",
+            "Acc  10/04/2009User2      Standard user."
+        }
+
+        ''' <summary>
+        ''' Create a new file with TestData
+        ''' </summary>
+        ''' <param name="TestBase">Object to manage temporary Files</param>
+        ''' <param name="TestData">Data to be written to file</param>
+        ''' <param name="PathFromBase">Optional additional subdirectories that file will be created under</param>
+        ''' <param name="TestFileName">Optional Filename, if none a random one based on TestName will be created</param>
+        ''' <returns>Full Path to New File</returns>
+        Private Shared Function CreateTestFile(TestBase As TextFieldParserTests, TestData As String, PathFromBase As String, TestFileName As String, <CallerMemberName> Optional memberName As String = Nothing, <CallerLineNumber> Optional lineNumber As Integer = 0) As String
```

Comment:
This method looks identical to the method in `FileIOTests`. Please create a common helper method that both classes can use.

Reply:
It is identical but TestBase.GetTestFilePath is protected and I can't call it from helper. Still working on LongPath, your suggestion is failing because on NETFX x86 the wrong error is being thrown. I think this is a bug in IO.CreateDirectory. I could look for both errors but then not sure the test is worth doing.

Reply:
Would it be worth creating an abstract base class for these two test classes and moving the method there?
```VB
Public MustInherit Class FileIOTestBase
    Inherits FileCleanupTestBase
    Protected Shared Function CreateTestFile(...)
        ...
    End Function
End Class

Public NotInheritable Class FileIOTests
    Inherits FileIOTestBase
    ...
End Class
```

Reply:
Done